Source/Core/DolphinQt/Achievements/AchievementProgressWidget.h

```diff
@@ -21,14 +23,13 @@ class AchievementProgressWidget final : public QWidget
   Q_OBJECT
 public:
   explicit AchievementProgressWidget(QWidget* parent);
-  void UpdateData();
+  void UpdateData(bool clean_all);
+  void UpdateData(std::set<AchievementManager::AchievementId> update_ids);
 
 private:
-  QGroupBox* CreateAchievementBox(const rc_api_achievement_definition_t* achievement);
-  QString GetStatusString(u32 achievement_id) const;
-
   QGroupBox* m_common_box;
   QVBoxLayout* m_common_layout;
+  std::map<AchievementManager::AchievementId, AchievementBox*> m_achievement_boxes;
```

Comment:
Can this be changed to an appropriate smart pointer, please, to avoid the explicit use of `new`/`delete`.